### PR TITLE
Added /getSlashes #628

### PR DESCRIPTION
**Pull request by nblogist**
## Description
Added /getSlashes route, to get slashes for the last 30 days, w.r.t stash provided or not.

## Motivation and Context
Implement /getSlashes route #628

## Have proper tags been added (for bug, enhancement, breaking change)?
- [ ] yes

## Does this PR affect any server routes?
- [ ] yes, and they are tested: [enter the % coverage here]
- [ ] yes, and they are not tested: [enter the % coverage here]
- [x] yes, but I did not run tests
- [ ] no

### CODE REVIEW

**Comment by drewstone** [comment]
Looks like there are conflicts here and there are no tests. Please at least stub out the route tests.

File: server/routes/getSlashes.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    endDate = new Date(); // today
+  }
+  /**
+   * {
+        includeAddresses: [ 'Alice' ],
+        data: { kind: 'slash', validator: 'Alice', amount: '10000' },
+        blockNumber: 10
+    }
+   */
+  if (req.query.stash) {
```
Comment (drewstone) [inline server/routes/getSlashes.ts:34]: You're not reusing enough code imo. You should create the query string based on `if (req.query.stash)` but then reuse the same querying logic to execute that string. This way you don't need to have to same code repeated twice.

All your routes should adopt this paradigm.

Comment (nblogist) [reply server/routes/getSlashes.ts]: Thank you for letting me know this, I'll make sure I use this approach now on!

File: server/routes/getSlashes.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      ['created_at', 'ASC']
+    ],
+    include: [ { model: models.ChainEventType } ]
+  });
+  return slashes;
+};
+const getSlashes  = async (models, req: Request, res: Response, next: NextFunction) => {
+  const slashes = await slashQuery(models, req, res, next);
+
+  if (slashes) { if (!slashes.length) return []; } else return [];
```
Comment (drewstone) [inline server/routes/getSlashes.ts:49]: it shouldn't be return [], it should return a properly formatted REST response with the result being [].

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks good, all tests should cover ALL error cases as well. So calling these routes without proper arguments should be included in the tests. They're very quick and easy to run and check since oyou jus have to fail to provide the correct values.

**Comment by drewstone** [comment]
@nblogist I've updated the return value.

What is left on this PR to complete?

**Comment by nblogist** [comment]
Due to the same [reason](https://github.com/hicommonwealth/commonwealth/pull/606#issuecomment-688263517): This PR is pointed to **_tmp_furqan-555-556-594_**

So, **_furqan-issue-628_**  goes into **_tmp_furqan-555-556-594_** , 

then **_tmp_furqan-555-556-594_** goes to **_staking-ui_**

File: server/routes/getExposureOverTime.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    };
+  }
+
+  const stakeOverTime = await models.HistoricalValidatorStatistic.findAll({
+    where,
+    order: [
+      ['created_at', 'ASC']
+    ],
+    attributes: ['stash', 'exposure', 'block'],
+    include: [ { model: models.ChainEventType } ]
```
Comment (drewstone) [inline server/routes/getExposureOverTime.ts:52]: can you describe what adding this model is doing and how you are sure this gets you the data you expect?

File: server/routes/getOffences.ts [hunk 4]
```diff
@@ -38,7 +38,8 @@ const getOffences = async (models, req: Request, res: Response, next: NextFuncti
     ]
   });
   // No Offences
-  if (!offences.length) return [];
+  if (offences) { if (!offences.length) return []; } else { return []; }
```
Comment (drewstone) [inline server/routes/getOffences.ts:41]: This is incorrect, you can't return a non-RESTFUL response

Comment (drewstone) [reply server/routes/getOffences.ts:41]: Please just use: return `res.json({ status: 'Success', result: offences || [] });`

File: server/routes/getSlashes.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      ['created_at', 'ASC']
+    ],
+    include: [ { model: models.ChainEventType } ]
+  });
+  return slashes;
+};
+const getSlashes  = async (models, req: Request, res: Response, next: NextFunction) => {
+  const slashes = await slashQuery(models, req, res, next);
+
+  return res.json({ status: 'Success', result: { slashes: slashes || [] } });
```
Comment (drewstone) [inline server/routes/getSlashes.ts:49]: this is an example I added afaik that you can use for the above.

**Review by drewstone — CHANGES_REQUESTED** [review]

File: server/routes/getOffences.ts (hunk 4 above)
Comment (nblogist) [reply server/routes/getOffences.ts:41]: Alright!

File: server/routes/getExposureOverTime.ts (hunk 3 above)
Comment (nblogist) [reply server/routes/getExposureOverTime.ts:52]: This was an unnecessary join, I have made the changes in https://github.com/hicommonwealth/commonwealth/pull/606/commits/8283f4f320fa9dbd99307a95a9803296b6415bd4 at **_tmp_furqan-555-556-594_**

**Comment by drewstone** [comment]
@nblogist please fix up the merge conflicts now that the other PR is merged.

**Pull request merged**